src/api/GenericApi.ts

```diff
@@ -46,10 +51,30 @@ export interface IGradePostAPI {
   grade: number;
 }
 
-export type APIObjectGet = IStudyProgramsAPI | ISubjectAPI | IStudentDetailsAPI[] | ICourseGetAPI | null;
-export type APIObjectPost = IStudyProgramsAPI | ISubjectAPI | IStudentDetailsAPI[] | ICoursePostAPI | IGradePostAPI  | null;
+export interface APIObjectGet {
```

Comment:
Caveat: there's no API definitions so these types most likely are wrong and supplied only as an example